lua/packer/plugin_types/git.lua

```diff
@@ -331,6 +332,26 @@ git.setup = function(plugin)
     end)()
     return r
   end
+
+  plugin.get_rev = function ()
```

Comment:
Nitpick, but it feels like the main logic here could be separated out into a function not associated with a specific plugin, then the `plugin.get_rev` becomes a simpler function calling the refactored `get_rev` function with the plugin's name and install location. This might be preferable because getting the revision of a plugin is something we do in several places, so reusing the same function simplifies the overall logic.